### ISSUE TITLE
`kafka_consumed_messages_total` topic label

### ISSUE THREAD

**Issue by jpds** [vectordotdev/vector#21330]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Use Cases

The `vector_kafka_consumed_messages_total` metric should have a `topic` label as there may be multiple topics configured for a source.

**Comment by jszwedko** [comment]
Agreed. Would you also want this at the partition level? Or is just topic sufficient?
